eta/core/data.py

```diff
@@ -853,35 +853,24 @@ def from_dict(cls, d, record_cls=None):
         Args:
             d: a DataRecords dictionary
             record_cls: an optional records class to use when parsing the
-                records dictionary. If not provided, the DataRecord dictionary
-                must define it
+                records dictionary. If not provided, the DataRecords dictionary
+                you are loading must have been written in reflective mode
 
         Returns:
             a DataRecords instance
         '''
-        rc = record_cls or d.get(cls._ELE_CLS_FIELD, None)
-        if rc is None:
-            raise DataRecordsError(
-                "Need record_cls to parse the DataRecords dictionary")
+        if record_cls is None:
+            record_cls_str = d.get(cls._ELE_CLS_FIELD, None)
+            if record_cls_str is None:
+                raise DataRecordsError(
+                    "Your DataRecords does not have its '%s' attribute "
+                    "populated, so you must manually specify the `record_cls` "
+                    "to use when loading it" % cls._ELE_CLS_FIELD)
+            record_cls = etau.get_class(record_cls_str)
 
         return DataRecords(
-            record_cls=rc,
-            records=[rc.from_dict(r) for r in d[cls._ELE_ATTR]])
-
-    @classmethod
-    def from_json(cls, json_path, record_cls=None):
```

Comment:
Noting that (1) I believe this was created before the `from_json` handled named args robustly (not important, but just noting) and (2) how do we document this case -->  the `from_json` that is now used does not explicitly have the `record_cls` argument and hence how is the new developer going to know this can be called in this way.